### ISSUE TITLE
Proposal: Warn on eval "$(run-something)"

### ISSUE THREAD

**Issue by akshaymankar** [koalaman/shellcheck#1740]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash

set -e

eval "$(some-command-which-could-fail)"
```

#### Here's what shellcheck currently says:

Nothing

#### Here's what I wanted or expected to see:

Instead of using eval directly on result of `"$(some-command-which-could-fail)"`, it would be better if it was written like this:

```sh

#!/bin/bash

set -e

foo="$(some-command-which-could-fail)"
eval "${foo}"
```

Best,

Akshay && @alex-slynko
